js/stats.js

```diff
@@ -0,0 +1,74 @@
+function parse_tsv(s, f) {
+  var ix_end = 0;
+  for (var ix=0; ix<s.length; ix=ix_end+1) {
+    ix_end = s.indexOf('\n', ix);
+    if (ix_end == -1) {
+      ix_end = s.length;
+    }
+    var row = s.substring(ix, ix_end).split('\t');
+    f(row);
+  }
+}
+
+function insertPoint(trace, x, y) {
+  if (/\d+/.exec(y)) {
+    trace.x.push(x);
+    trace.y.push(y);
+  }
+}
+
+function tsvListener() {
+  timeline = document.getElementById('timeline');
+
+  tIssued = { type: "scatter", name: "Issued", x:[], y:[],
+              yaxis: 'y2', fill: "tozeroy", showlegend: false,
+              line: { color: '#2a7ae2' } }
+  tActive = { type: "scatter", name: "Certificates Active", x:[], y:[],
+              line: { color: '#fa3a12' } }
+  tFqdn = { type: "scatter", name: "Fully-Qualified Domains Active", x:[], y:[] }
+  tRegDom = { type: "scatter", name: "Registered Domains Active", x:[], y:[],
+              marker: { symbol: "diamond" } }
+
+  dateFormat = /\d{4}-\d{2}-\d{2}/;
+  numFormat = /\d+/;
+
+  parse_tsv(this.responseText.trim(), function(row){
```

Comment:
Rather than trimming here, it would probably be cleaner to make parse_tsv robust to empty lines.